src/library/Yi/Paths.hs

```diff
@@ -52,7 +51,7 @@ getConfigPath = getCustomConfigPath getConfigDir
 -- | Given an action that retrieves config path, and a path relative to it,
 -- this function joins the two together to create a config file path.
 getCustomConfigPath :: MonadBase IO m => m FilePath -> FilePath -> m FilePath
-getCustomConfigPath cd fp = (</> fp) `liftM` cd
+getCustomConfigPath cd fp = (</> fp) <$> cd
```

Comment:
Remember that `<$>` was introduced in Prelude only in GHC 7.10, so you may have to use an import with required ifdefs to maintain backward compatibility with GHC 7.8 (which is still supported by Yi).